File: MyApp.Domain/Entities/Comment.cs

namespace MyApp.Domain.Entities
{
    public class Comment
    {
        public int Id { get; set; }

        public string Text { get; set; } = string.Empty;

        public int ContentId { get; set; }
        public Content? Content { get; set; }

        public int UserId { get; set; }
        public User? User { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: Frontends/MyApp.WebUI/src/app/services/comment.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})

export class CommentService {
  private apiUrl = 'http://localhost:5001/api/Comment';

  constructor(private http: HttpClient) { }

  getCommentsByContentId(contentId: number): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/content/${contentId}`);
  }

  addComment(contentId: number, text: string): Observable<any> {
    return this.http.post(`${this.apiUrl}`, {
      contentId,
      text
    });
  }

  deleteComment(id: number): Observable<any> {
    return this.http.delete(`${this.apiUrl}/${id}`);
  }

  getCommentsByUserId(userId: number): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/user/${userId}`);
  }

  updateComment(id: number, text: string): Observable<any> {
    return this.http.put(`${this.apiUrl}`, {
      id,
      text
    });
  }
}